kratos/includes/model_part.h

```diff
@@ -574,6 +604,110 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     }
 
 
+    ///@}
+    ///@name MasterSlaveConstraints
+    ///@{
+
+    SizeType NumberOfMasterSlaveConstraints() const
+    {
+        return mMasterSlaveConstraints.size();
+    }
+
+    MasterSlaveConstraintContainerType& MasterSlaveConstraints()
+    {
+        return mMasterSlaveConstraints;
+    }
+
+    const MasterSlaveConstraintContainerType& MasterSlaveConstraints() const 
```

Comment:
constraints are not in mesh...so we simply cannot add it to a mesh.

if your case is about adding constraint to the mesh instead of to the modelpart directly, than you may be right...simply i sort of consider the meshes as legacy (my personal opinion on this though...)